File: Sistemas de ecuaciones no lineales/Newton/Newton_Mod/Julia/newton_mod.jl
#using LinearAlgebra
#En la libería estándar existe en LinearAlgebra la función norma que hace lo mismo. Meh.
function norma(A::Array{Float64})
    ∑ = 0.0

    for i in 1:length(A)
        ∑ += (A[i] ^ 2)
    end

    return √∑

end
#Usando despazamientos simultaneos
function newton_mod(X::Array{Float64}, F::Array{Function}, δF::Array{Function}, tolerancia::Float64, iteraciones::Int64)

    X_km1 = copy(X)
    X_k = copy(X)
    F_X_k_ = zeros(Float64, length(F), 1)
    δF_X_k_ = zeros(Float64, length(F), 1)
    e_a = 1.0
    k = 0

    print("X = ")
    println(X_km1)

    while true

        X_km1 = copy(X_k)

        for i ∈ 1:length(F)
            F_X_k_[i] = F[i](X_km1)
        end

        print("F(X) = ")
        println(F_X_k_)

        for i ∈ 1:length(δF)
            δF_X_k_[i] = δF[i](X_km1)
        end

        print("δF(X) = ")
        println(δF_X_k_)

        e_a = norma(F_X_k_)

        println(e_a)

        if  e_a <= tolerancia
            break
        end
        if k >= iteraciones
            break
        end

        for i ∈ 1:length(X_k)
            X_k[i] = X_km1[i] - ((F_X_k_[i])/(δF_X_k_[i]))
        end

        print("K = ")
        println(k)
        print("X^k = ")
        println(X_k)

        k += 1
    end

    return X_k

end

File: Sistemas de ecuaciones no lineales/Newton/Newton_Mod/C/newton_mod.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

typedef struct
{
    long double x;
    long double y;
    long double z;
}Punto;

long double norma(Punto X)
{
    long double s = 0.0;

    s += (X.x * X.x);
    s += (X.y * X.y);
    s += (X.z * X.z);

    return sqrt(s);
}

Punto newton_mod(Punto X, long double (**F)(Punto X), long double (**F_d)(Punto X), long double tolerancia, int iteraciones)
{ 
    Punto X_k = X;
    Punto F_X_k, dF_X_k;
    long double e_a;
    int k = 0;

    while(1)
    {

        F_X_k.x = F[0](X_k);
        F_X_k.y = F[1](X_k);
        F_X_k.z = F[2](X_k);

        dF_X_k.x = F_d[0](X_k);
        dF_X_k.y = F_d[1](X_k);
        dF_X_k.z = F_d[2](X_k);

        e_a = norma(F_X_k);

        printf("|%-4d|%-11.10LF|%11.10LF|%11.10LF|%-10.10LF|\n", k, X_k.x, F_X_k.x, dF_X_k.x, e_a);
        printf("|    |%-11.10LF|%11.10LF|%11.10LF|\n", X_k.y, F_X_k.y, dF_X_k.y);
        printf("|    |%-11.10LF|%11.10LF|%11.10LF|\n\n", X_k.z, F_X_k.z, dF_X_k.z);

        if (e_a <= tolerancia) break;
        if (k >= iteraciones) break;

        X_k.x = X_k.x - ((F_X_k.x)/(dF_X_k.x));
        X_k.y = X_k.y - ((F_X_k.y)/(dF_X_k.y));
        X_k.z = X_k.z - ((F_X_k.z)/(dF_X_k.z));

        k++;
    }

    return X_k;
}